File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.rs
impl Solution {
    pub fn erase_overlap_intervals(mut intervals: Vec<Vec<i32>>) -> i32 {
        intervals.sort();

        let mut res = 0;
        let mut prev_end = intervals[0][1];

        for interval in intervals.iter().skip(1) {
            let (start, end) = (interval[0], interval[1]);
            if start < prev_end {
                res += 1;
                prev_end = prev_end.min(end);
            } else {
                prev_end = end;
            }
        }

        res
    }
}

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& in) {
        sort(in.begin(), in.end(), [](const vector<int> &a, const vector<int> &b){
            return a[1] < b[1]; 
        });


        int cnt = 0, prevEnd = in[0][1];

        for(int i=1 ; i<in.size() ; i++){
            if(in[i][0] < prevEnd){
                cnt++;
            } else {
                prevEnd = in[i][1];
            }
        }

        return cnt;
    }
};